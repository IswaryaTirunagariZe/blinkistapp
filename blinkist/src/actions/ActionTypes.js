export const ADD_TO_LIBRARY = 'ADD_TO_LIBRARY'
export const ADD_BOOK_TO_LIST = 'ADD_BOOK_TO_LIST'
export const FINISH_READING = 'FINISH_READING'
export const READ_AGAIN = 'READ_AGAIN'
export const SEARCH_BOOKS = 'SEARCH_BOOKS'
export const GET_ALL_BOOKS = 'GET_ALL_BOOKS'