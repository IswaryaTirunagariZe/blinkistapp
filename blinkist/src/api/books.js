export const allBooks=[{  
  'id': 1,  
  'imageUri':"https://source.unsplash.com/random", 
  'bookTitle': 'Book1',   
  'authors': 'Book1 authors' ,
  'category':'Entertainment',
  'inLibrary' :true,
  'isReadFinished':false,
  'isReading':true,
},
{  
    'id': 2, 
    'imageUri':"https://source.unsplash.com/random",  
    'bookTitle': 'Book2',   
    'authors': 'Book2 authors' ,
    'category':'Entertainment',
    'inLibrary' :false,
    'isReadFinished':true,
    'isReading':false
  },
  {  
    'id': 3,  
    'imageUri':"https://source.unsplash.com/random", 
    'bookTitle': 'Book3',   
    'authors': 'Book3 authors',
    'category':'Entertainment',
    'inLibrary' :true,
    'isReadFinished':true,
    'isReading':false
  },
  {  
    'id': 44,  
    'imageUri':"https://source.unsplash.com/random", 
    'bookTitle': 'Book44',   
    'authors': 'Book3 authors',
    'category':'Entertainment',
    'inLibrary' :true,
    'isReadFinished':true,
    'isReading':false
  },
  {  
      'id': 4,  
      'imageUri':"https://source.unsplash.com/random", 
      'bookTitle': 'Book4',   
      'authors': 'Book4 authors',
      'category':'Trending',
      'inLibrary' :true,
      'isReadFinished':true,
      'isReading':false
    },
    {  
      'id': 5,  
      'imageUri':"https://source.unsplash.com/random", 
      'bookTitle': 'Book5',   
      'authors': 'Book5 authors',
      'category':'Trending',
      'inLibrary' :true,
      'isReadFinished':true,
      'isReading':false
    },
    {  
      'id': 6,  
      'imageUri':"https://source.unsplash.com/random", 
      'bookTitle': 'Book6',   
      'authors': 'Book6 authors',
      'category':'To Do',
      'inLibrary' :true,
      'isReadFinished':true,
      'isReading':false
    },
    {  
      'id': 7,  
      'imageUri':"https://source.unsplash.com/random", 
      'bookTitle': 'Book7',   
      'authors': 'Book7 authors',
      'category':'To Do',
      'inLibrary' :true,
      'isReadFinished':false,
      'isReading':true
    },
];