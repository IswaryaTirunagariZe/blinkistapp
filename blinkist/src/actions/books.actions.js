import * as actionsTypes from './ActionTypes';



export const getAllbooks = () => dispatch => {
  //const books=getBooksFromJson();
  dispatch(getBooks());
}
const getBooks = books => ({
  type: actionsTypes.GET_ALL_BOOKS,
  books
})
const addToBookLibrary = bookId => ({
  type: actionsTypes.ADD_TO_LIBRARY,
  bookId
})

const finishReading = bookId => ({
  type: actionsTypes.FINISH_READING,
  bookId
})
const readAgain = bookId => ({
  type: actionsTypes.READ_AGAIN,
  bookId
})
const addNewBookToList = book => ({
  type: actionsTypes.ADD_BOOK_TO_LIST,
  book
})

export const addToLibrary = (bookId) => dispatch => {
 dispatch(addToBookLibrary(bookId))
  }

export const libraryButtonActions = (buttonAction,bookId) => dispatch => {
  if(buttonAction==='FinishReading'){
    dispatch(finishReading(bookId))
  }
    if(buttonAction==='ReadAgain'){
    dispatch(readAgain(bookId))
  }
     }

export const addNewBook = (book) => dispatch => {
  dispatch(addNewBookToList(book))
    }