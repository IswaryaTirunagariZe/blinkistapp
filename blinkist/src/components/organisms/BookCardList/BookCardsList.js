import React from 'react'
import { connect } from 'react-redux'
import { addToLibrary } from '../../../actions/books.actions'
import Grid from '@material-ui/core/Grid';
import Container from '@material-ui/core/Container';
import BookCard from '../../molecules/BookCard/BookCard';
import Typography from '@material-ui/core/Typography';

const categeories = [
  {
    'id': 1,
    'cat': 'Entertainment'
  },
  {
    'id': 2,
    'cat': 'Trending'
  },
  {
    'id': 3,
    'cat': 'To Do'
  },
];
const classes={};

const BookCardsList = ({ books, addToLibrary }) => (
  <main>
    {categeories.map((categeories) => (
      <Container maxWidth="md">
        <Typography gutterBottom variant="h5" component="h2">
          {categeories.cat}
        </Typography>
        <Grid container spacing={4} gutterBottom >
           {books.filter(_books => _books.category === categeories.cat).map((book) => (
            <Grid item key={book.id} xs={12} sm={6} md={4}>
              <BookCard books={book} buttonAction="+ Add To Library"
                handleClick={() => addToLibrary(book.id)} classes={classes}></BookCard>
            </Grid>

          ))}
        </Grid>
      </Container>
   ))}

  </main>
);

const mapStateToProps = state => ({
  books: state.books
})

export default connect(
  mapStateToProps,
  { addToLibrary }
)(BookCardsList)