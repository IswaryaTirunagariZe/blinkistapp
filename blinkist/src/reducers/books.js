import { combineReducers } from 'redux'
import { FINISH_READING, ADD_TO_LIBRARY,GET_ALL_BOOKS } from '../actions/ActionTypes'
import api from '../api/getBooks'
import { allBooks } from '../api/books';

const getBooksFromJson = () => {  
  const response = api.get("/books").then(resp => {
    var data = resp.data;
    return response.data;
})
.catch(error => {
    console.log(error);
});
  
};

const initialState = {
  //books: [getBooksFromJson()]
  books:allBooks
}


const books = (state=initialState, action) => {
  switch (action.type) {
    case 'ADD_TO_LIBRARY':
      state.filter(_books => _books.id == action.bookId).map((book)=>
      { book.inLibrary=true;
        book.isReading=true;
        book.isReadFinished=false;
      })
      return [...state]
      case 'GET_ALL_BOOKS':
      return state 
      case 'FINISH_READING':
        state.filter(_books => _books.id == action.bookId).map((book)=>
        { book.isReadFinished=true;
          book.isReading=false;
        })
        return [...state] 
      case 'READ_AGAIN':
        state.filter(_books => _books.id == action.bookId).map((book)=>
        { book.isReading=true;
        book.isReadFinished=false})
      return [...state] 
      case 'ADD_BOOK_TO_LIST':
        state.push(action.book);
      return [...state]  
    default:
      return state.books
  }
}

export default combineReducers({
  books
})

