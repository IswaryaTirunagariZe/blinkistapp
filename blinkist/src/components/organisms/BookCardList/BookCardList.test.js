import { shallow, render } from "enzyme";
import BookCardsList from "./BookCardsList";
import { createStore, applyMiddleware } from 'redux'
import { Provider } from 'react-redux'
import reducer from '../../../reducers/books'
import { getAllbooks } from '../../../actions/books.actions'

describe("Book Cards List", () => {
  it("Renders Book Cards List", () => {
    const store = createStore(
        reducer
      )
    const wrapper = shallow(<Provider store={store}>
        <BookCardsList/></Provider>);
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
});