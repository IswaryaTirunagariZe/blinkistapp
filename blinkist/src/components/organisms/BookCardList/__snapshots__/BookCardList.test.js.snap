// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book Cards List Renders Book Cards List 1`] = `ShallowWrapper {}`;
